InnerEye/ML/Histopathology/datasets/base_dataset.py

```diff
@@ -0,0 +1,107 @@
+from pathlib import Path
+from typing import Any, Dict, Optional, Union
+
+import numpy as np
+import pandas as pd
+import torch
+from sklearn.utils.class_weight import compute_class_weight
+from torch.utils.data import Dataset
+
+
+class TilesDataset(Dataset):
+    """Base class for datasets of WSI tiles, iterating dictionaries of image paths and metadata.
+
+    :param TILE_ID_COLUMN: CSV column name for tile ID.
+    :param SLIDE_ID_COLUMN: CSV column name for slide ID.
+    :param IMAGE_COLUMN: CSV column name for relative path to image file.
+    :param PATH_COLUMN: CSV column name for relative path to image file. Replicated to propagate the path to the batch.
+    :param LABEL_COLUMN: CSV column name for tile label.
+    :param SPLIT_COLUMN: CSV column name for train/test split (optional).
+    :param TILE_X_COLUMN: CSV column name for horizontal tile coordinate (optional).
+    :param TILE_Y_COLUMN: CSV column name for vertical tile coordinate (optional).
+    :param TRAIN_SPLIT_LABEL: Value used to indicate the training split in `SPLIT_COLUMN`.
+    :param TEST_SPLIT_LABEL: Value used to indicate the test split in `SPLIT_COLUMN`.
+    :param DEFAULT_CSV_FILENAME: Default name of the dataset CSV at the dataset rood directory.
+    :param N_CLASSES: Number of classes indexed in `LABEL_COLUMN`.
+    """
+    TILE_ID_COLUMN: str = 'tile_id'
+    SLIDE_ID_COLUMN: str = 'slide_id'
+    IMAGE_COLUMN: str = 'image'
+    PATH_COLUMN: str = 'image_path'
+    LABEL_COLUMN: str = 'label'
+    SPLIT_COLUMN: Optional[str] = 'split'
+    TILE_X_COLUMN: Optional[str] = 'tile_x'
+    TILE_Y_COLUMN: Optional[str] = 'tile_y'
+
+    TRAIN_SPLIT_LABEL: str = 'train'
+    TEST_SPLIT_LABEL: str = 'test'
+
+    DEFAULT_CSV_FILENAME: str = "dataset.csv"
+
+    N_CLASSES: int = 1  # binary classification by default
```

Comment:
We're touching this class right now, making @daniel-castro available that his changes will need to be merged here